src/net.c

```diff
@@ -1048,6 +1061,88 @@ token_match(const uint8_t *a, size_t alen,
   return alen == blen && (alen == 0 || memcmp(a, b, alen) == 0);
 }
 
+int
+coap_client_delay_first(coap_session_t *session)
+{
+  if (session->type == COAP_SESSION_TYPE_CLIENT && session->doing_first) {
+#ifndef WITH_LWIP
+    if (session->doing_first) {
+      unsigned ref;
+      int timeout_ms = 5000;
+      /*
+       * Need to wait for first request to get out and response back before
+       * continuing.. Response handler has to clear doing_first if not an error.
+       */
+      ref = ++session->ref;
+      while (*(&session->doing_first) != 0) {
```

Comment:
For better readability, I suggest `session->doing_first != 0`

Reply:
Done